src/p_spec.c

```diff
@@ -2264,6 +2264,20 @@ boolean         levelFragLimit;      // Ty 03/18/98 Added -frags support
 int             levelFragLimitCount; // Ty 03/18/98 Added -frags support
 
 int             r_swirl;
+int             swirl_custom_rule;
+
+//51 = NUKAGE1 69 = FWATER1 73 = LAVA1 89 = BLOOD1 120 = RROCK05 136 = SLIME01 140 = SLIME05 144 = SLIME09
+boolean Swirl_CustomRule(int basepic, int val)
+{
+  return (((val >> 7) & 1) && basepic == 51)  ? true : false ||
+         (((val >> 6) & 1) && basepic == 69)  ? true : false ||
```

Comment:
AFAIUI, `basepic` depends on the number and ordering of flat lumps. That is, if a PWAD adds a number of flats and one of them is among the ones in the ANIMATED list, then the `basepic` values compared here wouldn't be valid anymore. So, in the end, this approach only works for sure as intended for IWADs - which is where the previous implementation already worked as well (i.e. all animated IWAD flats do look reasonable when swirling).